src/payment/coupon/CouponForm.jsx

```diff
@@ -33,16 +33,69 @@ export class CouponForm extends Component {
     this.props.removeCoupon(this.props.voucherId);
   }
 
+  renderInvalidMessage() {
+    const { code, errorCode, intl } = this.props;
+    if (errorCode === null) {
+      return null;
+    }
+    let invalidMessage = null;
+    switch (errorCode) {
+      case 'empty_basket':
+        invalidMessage = intl.formatMessage(messages['payment.coupon.error.empty_basket'], {
+          code,
+        });
+        break;
+      case 'already_applied_voucher':
+        invalidMessage = intl.formatMessage(messages['payment.coupon.error.already_applied_voucher'], {
+          code,
+        });
+        break;
+      case 'code_does_not_exist':
+        invalidMessage = intl.formatMessage(messages['payment.coupon.error.code_does_not_exist'], {
+          code,
+        });
+        break;
+      case 'code_expired':
+        invalidMessage = intl.formatMessage(messages['payment.coupon.error.code_expired'], {
+          code,
+        });
+        break;
+      case 'code_not_active':
+        invalidMessage = intl.formatMessage(messages['payment.coupon.error.code_not_active'], {
+          code,
+        });
+        break;
+      case 'code_not_available':
+        invalidMessage = intl.formatMessage(messages['payment.coupon.error.code_not_available'], {
+          code,
+        });
+        break;
+      case 'code_not_valid':
+        invalidMessage = intl.formatMessage(messages['payment.coupon.error.code_not_valid'], {
+          code,
+        });
+        break;
+      default:
+        invalidMessage = intl.formatMessage(messages['payment.coupon.error.unknown']);
```

Comment:
Should we log here?

Reply:
Good point!  Yeah, maybe... let me add something.